### ISSUE TITLE
Poor performance during editing

### ISSUE THREAD

**Issue by RazrFalcon** [intellij-rust/intellij-rust#1572]
I'm trying to use IDEA instead of Sublime Text as a Rust IDE, but I encountering huge lags during code input.

Here is a screenshot: https://itmages.ru/image/view/5984673/ae426f8c

This is the latest IDEA with bundled JVM. On the right side you can see *conky* showing CPU usage per core. Under the IDEA window, there is KDE system monitor showing that it's IDEA eating all the CPU.

The lags appear only during code input. Go to definition, highlighting, code completion, etc. works fine and fast. There are no lags during Java projects editing, so it's probably a plugin bug.

I'm not familiar with IDEA and Java itself, so I have no idea how to profile the plugin.

**Comment by vlad20012** [comment]
Hi, @RazrFalcon. Thank for the report!
Could you please tell the Rust project and the particular file/function you editing? Or does it not matter which project to edit?

**Comment by RazrFalcon** [comment]
Project itself doesn't matter. I have the same issues even in a new project that has only 100 LOC.

**Comment by vlad20012** [comment]
Is "Use cargo check to analyze code" disabled?
![image](https://user-images.githubusercontent.com/3221931/28937886-fc374162-7894-11e7-81d2-0d29e906eab5.png)

**Comment by RazrFalcon** [comment]
Yes. I have the same setting as in the screenshot. Except I'm using Rust 1.19

**Comment by vlad20012** [comment]
Then only type inference/resolve remains. It is not yet well-optimized, but we are working on it!

**Comment by RazrFalcon** [comment]
I see. Thanks.

**Comment by SanyaPapkins** [comment]
Работяги, тут такое дело. Не могу найти вообще в настройках Rust, это нормально?

**Comment by Undin** [comment]
@SanyaPapkins maybe Rust plugin is not installed/enabled in your IDE at all? You can check it via `Preferences | Plugins` settings
